appbuilder/services/livesync/livesync-service.ts

```diff
@@ -0,0 +1,102 @@
+///<reference path="../../../.d.ts"/>
+"use strict";
+
+import { exportedPromise } from "../../../decorators";
+
+export class ProtonLiveSyncService implements IProtonLiveSyncService {
+	private excludedProjectDirsAndFiles = ["app_resources", "plugins", ".*.tmp", ".ab"];
+
+	private get $liveSyncServiceBase(): ILiveSyncServiceBase {
+		return this.$injector.resolve("liveSyncServiceBase");
+	}
+
+	constructor(private $devicesService: Mobile.IDevicesService,
+		private $errors: IErrors,
+		private $mobileHelper: Mobile.IMobileHelper,
+		private $options: ICommonOptions,
+		private $fs: IFileSystem,
+		private $injector: IInjector,
+		private $project: any,
+		private $logger: ILogger) { }
+
+	@exportedPromise("liveSyncService")
+	public livesync(deviceIdentifiers: IDeviceLiveSyncInfo[], projectDir: string, filePaths?: string[]): IFuture<void> {
+		return (() => {
+			this.$project.projectDir = projectDir;
+			let deviceInfos: any = _.map(deviceIdentifiers, d => {
```

Comment:
You could add 
`device?: Mobile.IDevice` to `IDeviceLiveSyncInfo` interface and use the same interface as a return type here, instead of `any`